Lib/test/test_subprocess.py

```diff
@@ -2169,6 +2170,52 @@ def test_swap_fds(self):
         self.check_swap_fds(2, 0, 1)
         self.check_swap_fds(2, 1, 0)
 
+    def _check_swap_std_fds_with_one_closed(self, from_fds, to_fds):
+        saved_fds = self._save_fds(range(3))
+        try:
+            for from_fd in from_fds:
+                with tempfile.TemporaryFile() as f:
+                    os.dup2(f.fileno(), from_fd)
+
+            for fd in range(3):
+                if fd not in from_fds:
+                    os.close(fd)
+                    break
+
+            arg_names = ['stdin', 'stdout', 'stderr']
+            kwargs = {}
+            for from_fd, to_fd in zip(from_fds, to_fds):
+                kwargs[arg_names[to_fd]] = from_fd
+
+            code = textwrap.dedent(r'''
+                import os, sys
+                skipped_fd = int(sys.argv[1])
+                for fd in range(3):
+                    if fd != skipped_fd:
+                        os.write(fd, str(fd).encode('ascii'))
+            ''')
+
+            skipped_fd = next(fd for fd in range(3) if fd not in to_fds)
```

Comment:
I believe this works, but it feels like something using sets might be easier to understand?

`({0, 1, 2} - set(to_fds)).pop()` perhaps?